rfcs/2021-01-22-spatial-system-rfc.md

```diff
@@ -0,0 +1,204 @@
+
+2. **Padding:** An area around the content's border and its child elements.
+
+3. **Border:** A stroke/line that goes around the padding & content.
+
+4. **Margin:** An area around the content's border and it's neighbouring element.
+
+![The Box Model](./images/spatial/box-model.png)
+
+### **What is a pt?**
+A point (`pt`) is a measurement of space that is dependent on screen resolution. Simply put, at a `1x` resolution (or `@1x`). That is,
+```
+1pt = 1px
+```
+>**📝Note:** It is always recommended to design in **`1x`** and then exporting your assets to different sizes (`@2x`, `@3x`, etc.) depending upon the use case.
+
+### **Space ≠ Grid**
+Industry generally considers space as grids, but these two are different things. Spaces are used to make grids (i.e. grids/containers/etc. are built using spaces).
+
+### **8pt grid system**
+Before diving into the **8pt grid system** we need to consider few things where this spatial system will be used. We'll use this in overall layout for `components` (molecules, organisms, templates) as well as on the `typography`.
+
+>**📝Note:** This system will used across mobiles, tablets & desktops.
+
+### **But why 8pt grid?**
+There is no right or wrong answer here, but there are few reasons for choosing 8pt over all others (`5pt`, `6pt`, `7pt`...)
+- It scales perfectly in all type of screen displays, be it Android or iOS (`@0.75` or `@1.5`).
+- Apple & Google advice to use the same, without stating too many reasons though.
+- It's simple for us to do the math. The `4` & `8` are simpler to calculate and they are flexible enough to setup a scale.
+
+### **Base Unit**
+We'll start with the base unit, this will allow us to create a scale of all the sizes in your spatial system. For us this also means using multiples of `x8` to define dimensions, **padding**, and **margin** of elements. But for our use-case multiple of `x8` won't work, as multiples of 8 can go too high (prctically) to work at a system-level.
+
+### **What system will work for Razorpay?**
+We should consider few cases that will make the system flexible for us. Few things to consider here are the **user needs**.
+
+As a merchant (user) do they want the interface to be spacious with large fonts and limited number of actions? Absolutely not, we need something that doesn't look *too-dense* or *too-spacious*. Therefore we'll consider **`4pt base unit`** which will give us the flexibility to scale the UI in multiples of `x4`.
+
+We also need to focus on the number of variables we'll have in our scale. These are the variables/tokens developers will use to while defining a grid layout.
+
+### **How to apply a spatial system?**
+Applying a system can come in the form of `padding`, `margins`, `width` & `height`. So there are two things we can control here *main sizing* & *internal element sizing*.
+
+- **Element-first (strict element sizing):** In this type the size takes the priority. For examples *buttons*, *form inputs*, *etc*.
```

Comment:
If we have fixed sizing then how will we scale it across devices?
* User's device font size will increase font size of then element and hence because we have fixed sizing the text will get truncated or get overflowed of the element?
* Zooming/magnifying the page on web. how will that affect this?